### PR TITLE
[#3730] github actions ci

### LINKED ISSUE #3730: Introduce GitHub Actions for CI
### Feature Description

Switching from Travis CI to GitHub Actions provides an immediate solution to the very slow CI process.

This Issue and linked PR would not disable Travis CI, but enable GitHub Actions in addition, so that the results can be evaluated and compared later on.

### Problem Description

The current Travis CI process is very slow, in part due to CI processing queuing up for a lot of time.
Build minutes are also limited as announced last months (#3541) . In comparison, GitHub offers unlimited build minutes for public repositories ([source](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions))

This is of particular high risk as currently, PRs get regulary merged without the CI process finishing, posing security and stability risks (of the last 50 commits to development branch, none passed the tests, probably because most didn't even start) (see #3711 )

### Proposed Change

Enabling and integrating GitHub Actions is a fast and streamlined solution. It also opens up a whole lot of possibilities due to the full GitHub integration, like disabling merging for PRs whose tests failed.

In addition, the tests for JDK8 and JDK11 run fully parallel and finished nearly twice as fast as the current Travis CI (total 16:09min vs 27:05min)

### Additional Information

[Example Build with GitHub Actions](https://github.com/ImisDevelopers/SORMAS-Project/actions/runs/410539795)

[Travis CI Build for comparison](https://travis-ci.org/github/hzi-braunschweig/SORMAS-Project/builds/748367014)

[GitHub Actions Documentation](https://docs.github.com/en/free-pro-team@latest/actions)

### PR DESCRIPTION
**Pull request by JonasCir**
Closes #3730

### CODE REVIEW

**Comment by Seal33** [comment]
@markusmann-vg @MartinWahnschaffeSymeda 

It would be really nice if this PR could be reviewed and merged soon, to accelerate CI processes of ongoing and future PRs.

As mentioned in the issue description this will **not** disable the current Travis CI, but just offer a possibly faster alternative :)

**Review by tbroyer — COMMENTED** [review]
You may want to cache the ~/.m2/repository between builds, keyed on a hash of all the pom.xml files (also `serverlibs.pom`, would be really great to rename it to `pom.xml` BTW; see #3728).

**Comment by JonasCir** [comment]
@tbroyer Thanks for the review much appreciated! I take care of your comments ASAP

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Build with Maven
+        run: |
+          cd sormas-base
+          mvn install -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
```
[the remark this answers is not in the document]
Comment (JonasCir) [reply .github/workflows/ci.yml]: Thanks fixed!

**Comment by JonasCir** [comment]
Thanks @tbroyer Can you take another look?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
```
Comment (tbroyer) [inline .github/workflows/ci.yml:29]: The `hashFiles` should also include the `serverlibs.pom` (either as `hashFiles('**/pom.xml', '**/serverlibs.pom')` or `hashFiles('**/pom.xml', '**/*.pom')`), until it's eventually renamed…

Not sure if the `${{ matrix.java }}` should be part of the key too; technically the dependencies could be different due to Maven profiles. Worst case will be one build is cached and the other one needs to download the few dependencies that differ, but still downloads them every time it runs!
```suggestion
          key: ${{ runner.os }}-java-${{ matrix.java }}-m2-${{ hashFiles('**/pom.xml', '**/*.pom') }}
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    branches: [ development ]
+  pull_request:
+    branches: [ development ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ 1.8, 1.11 ]
```
[the remark this answers is not in the document]
Comment (JonasCir) [reply .github/workflows/ci.yml:17]: thanks

File: .github/workflows/ci.yml (hunk 2 above)
Comment (JonasCir) [reply .github/workflows/ci.yml:29]: Good catch with the matrix key! I will include `'**/*.pom` as suggested and remove it once `serverlibs` is renamed

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build with Maven
+        run: |
+          cd sormas-base
+          mvn verify -B -ntp
```
[the remark this answers is not in the document]
Comment (JonasCir) [reply .github/workflows/ci.yml:34]: yes

**Review by tbroyer — APPROVED** [review]

**Comment by JonasCir** [comment]
Final results are in:
15 Minutes for Matrix build of  Java 8 + 11 with hot caches
https://github.com/ImisDevelopers/SORMAS-Project/actions/runs/413251902

**Comment by JonasCir** [comment]
@MartinWahnschaffeSymeda Updated to Java 11 only and removed travis

Build times:  14m 51s

**Pull request merged**